### ISSUE TITLE
(優先度SSR)小説を書くための独自フォーマットを作り、それをプリプロセスかけることで小説の提出用文字列を出力できるようにする

### ISSUE THREAD

**Issue by elipmoc** [elipmoc/novel-shared-ptr#3]
要件

- `//`や`/*  */`などでコメントアウトが出来ること
- `#define 主人公 らいパン粉`などの様に定数を定義できること
- PDF出力のためのタグ(画像挿入、改ページ等)などを認識できること
- 上記の特殊な記法全てを置き換え、取り除いた文字列を出力できること
- オプションでPDF出力のためのタグだけ取り除かないモードがあること
- コマンドラインから1アクションのコマンドで文字列出力が可能な形式であること

**Comment by yumetodo** [comment]
Rust製のmdbookとかそういうのは使わない方向でしょうか？

**Comment by elipmoc** [comment]
本Issueにて追加しました要件を満たせればなんでも構わないのですけど。
どうなんでしょうか？mdbookについてまだあまり良く分かりませんので……。

**Comment by yumetodo** [comment]
いや、mdbookにしろvuepagesにしろ、plugin機構があるのでそれを使うほうが0から作るより楽じゃないかという話でした。

なおコメントアウトについてはmarkdown系はHTMLのコメントをそのまま使えると思います。
